lib/crowbar/client/command/upgrade.rb

```diff
@@ -0,0 +1,50 @@
+#
+# Copyright 2016, SUSE Linux GmbH
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+module Crowbar
+  module Client
+    module Command
+      #
+      # Module for the upgrade command implementations
+      #
+      module Upgrade
```

Comment:
Just off the top of my head, does it not make it more easier to edit later on if we loop over a list/dictonary for the same? Since I see that most of the code is repeating here.